### PR TITLE
CONMAN-171: adding groupId to action

### PR DESCRIPTION
**Pull request by sumitbhanwala**
Moving the groupId setting in action destination setting from the destination config settings.

## Testing

Tested in staging and local environment as well

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage-messaging-sendgrid/__tests__/send-email.test.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ describe.each(['stage', 'production'])('%s environment', (environment) => {
     }
   }
 
-  const getDefaultMapping = (overrides?: any) => {
+  const getDefaultMapping = (groupId?: string, overrides?: any) => {
```
Comment (lew-gordon) [inline packages/destination-actions/src/destinations/engage-messaging-sendgrid/__tests__/send-email.test.ts:75]: Why wouldn't the groupId go into the overrides?

**Review by lew-gordon — COMMENTED** [review]
Just one comment to hopefully simplify the tests then I think it looks good!

File: packages/destination-actions/src/destinations/engage-messaging-sendgrid/__tests__/send-email.test.ts (hunk 1 above)
Comment (sumitbhanwala) [reply packages/destination-actions/src/destinations/engage-messaging-sendgrid/__tests__/send-email.test.ts:75]: You are right it can go there.

**Review by lew-gordon — APPROVED** [review]
LGTM!

**Review by marinhero — APPROVED** [review]

**Pull request merged**